ui/actions/SearchBottomSheet.tsx

```diff
@@ -0,0 +1,303 @@
+import { c, s } from '@/features/style'
+import BottomSheet, {
+  BottomSheetBackdrop,
+  BottomSheetScrollView,
+  BottomSheetTextInput,
+  BottomSheetView,
+} from '@gorhom/bottom-sheet'
+import { useCallback, useRef, useState } from 'react'
+import { Pressable, Text, ScrollView, View } from 'react-native'
+import { Heading } from '../typo/Heading'
+import { XStack, YStack } from '../core/Stacks'
+import { Button } from '../buttons/Button'
+import { CompleteRef, Item, Profile } from '@/features/pocketbase/stores/types'
+import { pocketbase, useUserStore } from '@/features/pocketbase'
+import { SimplePinataImage } from '../images/SimplePinataImage'
+import { Ionicons } from '@expo/vector-icons'
+import { router } from 'expo-router'
+import { Sheet } from '../core/Sheets'
+import { NewRef, NewRefStep } from './NewRef'
+
+const HEADER_HEIGHT = s.$10
+
+export default function BacklogBottomSheet() {
+  const [index, setIndex] = useState(0)
+
+  const isMinimised = index === 0
+  const backlogSheetRef = useRef<BottomSheet>(null)
+  const [addingTo, setAddingTo] = useState('')
+  const [step, setStep] = useState('')
+  const [newRefTitle, setNewRefTitle] = useState('')
+  const [initialStep, setInitialStep] = useState<NewRefStep>('')
+
+  const [searchTerm, setSearchTerm] = useState('')
+  const [results, setResults] = useState<CompleteRef[]>([])
+
+  const [refs, setRefs] = useState<CompleteRef[]>([])
+
+  const { user } = useUserStore()
+
+  const minSnapPoint = Math.min(refs.length, 4) * (s.$3 + s.$1) + s.$1 + HEADER_HEIGHT
+
+  const onAddRefToSearch = (r: CompleteRef) => {
+    setRefs((prevState) => [...prevState.filter((ref) => ref.id !== r.id), r])
+    setSearchTerm('')
+    updateSearch('')
+  }
+
+  const updateSearch = async (q: string) => {
+    const search = async () => {
+      let refsResults: CompleteRef[] = []
+      if (q === '') return []
+      refsResults = await pocketbase
+        .collection<CompleteRef>('refs')
+        .getFullList({ filter: `title ~ "${q}"` })
+      return refsResults
+    }
+
+    setSearchTerm(q)
+    const result = await search()
```

Comment:
I don't think we need to have an async function `search` defined inside here because the outer function `updateSearch` is also async